src/coreclr/jit/emitxarch.cpp

```diff
@@ -128,6 +155,146 @@ bool emitter::IsAVXInstruction(instruction ins) const
+        case INS_addsubpd:
+        case INS_rcpps:
+        case INS_rcpss:
+        case INS_rsqrtps:
+        case INS_rsqrtss:
+        case INS_psignb:
+        case INS_psignd:
+        case INS_psignw:
+        case INS_roundps:
+        case INS_roundss:
+        case INS_roundpd:
+        case INS_roundsd:
+        case INS_blendps:
+        case INS_blendpd:
+        case INS_blendvps:
+        case INS_pblendw:
+        case INS_pblendvb:
+        case INS_blendvpd:
+        case INS_ptest:
+        case INS_phaddw:
+        case INS_phsubw:
+        case INS_phaddd:
+        case INS_phsubd:
+        case INS_phaddsw:
+        case INS_phsubsw:
+        case INS_lddqu:
+        case INS_phminposuw:
+        case INS_mpsadbw:
+        case INS_pclmulqdq:
+        case INS_aesdec:
+        case INS_aesdeclast:
+        case INS_aesenc:
+        case INS_aesenclast:
+        case INS_aesimc:
+        case INS_aeskeygenassist:
+        case INS_vzeroupper:
+        case INS_vperm2i128:
+        case INS_vperm2f128:
+        case INS_vpblendd:
+        case INS_vblendvps:
+        case INS_vblendvpd:
+        case INS_vpblendvb:
+        case INS_vtestps:
+        case INS_vtestpd:
+        case INS_vmaskmovps:
+        case INS_vmaskmovpd:
+        case INS_vpmaskmovd:
+        case INS_vpmaskmovq:
+        case INS_andn:
+        case INS_blsi:
+        case INS_blsmsk:
+        case INS_blsr:
+        case INS_bextr:
+        case INS_rorx:
+        case INS_pdep:
+        case INS_pext:
+        case INS_bzhi:
+        case INS_mulx:
+#ifdef TARGET_AMD64
+        case INS_shlx:
+        case INS_sarx:
+        case INS_shrx:
+#endif
+        case INS_lfence:
+        case INS_mfence:
+        case INS_movnti:
+        case INS_prefetchnta:
+        case INS_prefetcht0:
+        case INS_prefetcht1:
+        case INS_prefetcht2:
+        case INS_sfence:
+        // Might need new INS_<INS_NAME>*suffix* instructions for these.
+        case INS_por:            // INS_pord, INS_porq.
+        case INS_pxor:           // INS_pxord, INS_pxorq
+        case INS_movdqa:         // INS_movdqa32, INS_movdqa64.
+        case INS_movdqu:         // INS_movdqu8, INS_movdqu16, INS_movdqu32, INS_movdqu64.
+        case INS_pand:           // INS_pandd, INS_pandq.
+        case INS_pandn:          // INS_pandnd, INS_pandnq.
+        case INS_vextractf128:   // INS_vextractf32x4, INS_vextractf64x2.
+        case INS_vextracti128:   // INS_vextracti32x4, INS_vextracti64x2.
+        case INS_vinsertf128:    // INS_vinsertf32x4, INS_vinsertf64x2.
+        case INS_vinserti128:    // INS_vinserti32x4, INS_vinserti64x2.
+        case INS_vbroadcastf128: // INS_vbroadcastf32x4, INS_vbroadcastf64x2.
+        case INS_vbroadcasti128: // INS_vbroadcasti32x4, INS_vbroadcasti64x2.
+        {
+            return false;
+        }
+        default:
+        {
+            break;
+        }
+    }
+
+    return IsAvx512OrPriorInstruction(ins);
```

Comment:
This is going to be dependent on `AVX512VL` being supported for many instructions.

Reply:
Does the other change resolve this issue for now? As we add new instructions, the logic in TakesEvexPrefix should evolve to handle newer cases.

Reply:
> The code still looks generally correct and works on my Intel Core i9-11900H (11th Gen Tiger Lake) and my AMD Ryzen 7950X (Zen4) CPU.
> 
> There is a concern around some of the EVEX checks and the new encoding stress switch in relation to `AVX512VL`.

@tannergooding Did the changes resolve all of your concerns or are there any other changes you're waiting on?

Reply:
I think so, but I'll need to give it another pass when I'm back in office near the end of next week.